zio-http/src/main/scala/zio/http/codec/internal/EncoderDecoder.scala

```diff
@@ -243,8 +318,39 @@ private[codec] object EncoderDecoder                   {
       } else if (jsonDecoders.length == 1) {
         jsonDecoders(0)(body).map { result => inputs(0) = result }
       } else {
-        ZIO.foreachDiscard(jsonDecoders.zipWithIndex) { case (decoder, index) =>
-          decoder(body).map { result => inputs(index) = result }
+        body.asMultipartFormStream.flatMap { form =>
+          form.fields.runForeach { field =>
+            indexByName.get(field.name) match {
+              case Some(idx) =>
+                flattened.content(idx) match {
+                  case BodyCodec.Multiple(schema, _, _) if schema == Schema[Byte] =>
+                    field match {
+                      case FormField.Binary(_, data, _, _, _)          =>
+                        inputs(idx) = ZStream.fromChunk(data)
+                      case FormField.StreamingBinary(_, _, _, _, data) =>
+                        inputs(idx) = data
+                      case FormField.Text(_, value, _, _)              =>
+                        inputs(idx) = ZStream.fromChunk(Chunk.fromArray(value.getBytes(Charsets.Utf8)))
+                      case FormField.Simple(_, value)                  =>
+                        inputs(idx) = ZStream.fromChunk(Chunk.fromArray(value.getBytes(Charsets.Utf8)))
+                    }
+                    ZIO.unit
+                  case _                                                          =>
+                    formFieldDecoders(idx)(field).map { result => inputs(idx) = result }
+                }
+              case None      =>
+                ZIO.fail(HttpCodecError.MalformedBody(s"Unexpected multipart/form-data field: ${field.name}"))
```

Comment:
This will log the `Option` which contains the field name, right?

Reply:
No this is the `Form` level name (not the codec) which is `String`